.github/workflows/pr-deploy.yaml

```
@@ -22,14 +22,34 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
-      - run: |
-          echo A user with write permission has commented on PR $NUMBER
-        env:
-          NUMBER: ${{ github.event.issue.number }}
-        if: steps.check.outputs.has-permission
+      - name: Checkout
+        uses: actions/checkout@v2 # no way of getting the correct ref from the issue event, hence the below
+        if: steps.check.outputs.has-permission && ${{ github.event.comment.body == '!deploy'}} # looks like this needs to go on every step. meh.
 
-      - run: |
-          echo User does not have write permission to repo, on PR $NUMBER
+      - name: Checkout Pull Request
+        run: hub pr checkout ${{ github.event.issue.number }} # github cli is pre installed
         env:
-          NUMBER: ${{ github.event.issue.number }}
-        if: "! steps.check.outputs.has-permission"
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        if: steps.check.outputs.has-permission && ${{ github.event.comment.body == '!deploy'}}
+
+      - name: Declare variables that we can share across steps
+        id: vars
+        run: |
+          GIT_SHORT="$(git rev-parse --short HEAD)"
+          echo "::set-output name=git_short::${GIT_SHORT}"
+          echo "::set-output name=fqdn_record::${GIT_SHORT}.posthog.cc"
+        if: steps.check.outputs.has-permission && ${{ github.event.comment.body == '!deploy'}}
```

I think this should be the below but I'm not sure.

```suggestion
        if: ${{ steps.check.outputs.has-permission && github.event.comment.body == '!deploy' }}
```

---

I _think_ you might be right, kinda wishing I could test this without pushing it up 🙃

---

Yeah it's not a great dev experience :(